src/constructor.jl

```diff
@@ -327,34 +327,35 @@ function construct_yaml_timestamp(constructor::Constructor, node::Node)
     m = parse(Int, mat.captures[5])
     s = parse(Int, mat.captures[6])
 
-    if mat.captures[7] === nothing
-        return DateTime(yr, mn, dy, h, m, s)
-    end
-
     ms = 0
+    
     if mat.captures[7] !== nothing
         ms = mat.captures[7]
         if length(ms) > 3
             ms = ms[1:3]
         end
         ms = parse(Int, string(ms, repeat("0", 3 - length(ms))))
     end
-
-    delta_hr = 0
-    delta_mn = 0
-
-    if mat.captures[9] !== nothing
-        delta_hr = parse(Int, mat.captures[9])
-    end
-
-    if mat.captures[10] !== nothing
-        delta_mn = parse(Int, mat.captures[10])
+    
+    if mat.captures[8] === nothing
+        return DateTime(yr, mn, dy, h, m, s, ms)
```

Comment:
Actually - before we merge, I wonder about this - it makes the function not type-stable. Should this also be a `ZonedDateTime`, but just with UTC?

Reply:
Good idea! Nice that codecov shows it isn't currently tested. Any idea what the test should be? The other tests are taken exactly form the spec ([here](https://yaml.org/spec/1.2.2/#24-tags)) so we'd have to make up a line that has time and date but no timezone.

I'm torn on what to do about the DateTime vs ZonedDateTime thing... I think DateTime is used a lot more, and it is useful even without timezones. `Dates.now()`, for example, shows my local system time. If I wrote a YAML document and specified times with no consideration for timezone, I don't know if I'd want to the read to assume it was all UTC. Unfortunately I don't think the spec makes a clear choice so we'll have to pick an implementation ourselves.